Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -13372,8 +13377,13 @@ MediaProducerMediaStateFlags WebPageProxy::reportedMediaState() const
     return internals().reportedMediaCaptureState | (internals().mediaState - MediaProducer::MediaCaptureMask);
 }
 
-void WebPageProxy::updatePlayingMediaDidChange(MediaProducerMediaStateFlags newState, CanDelayNotification canDelayNotification)
+void WebPageProxy::updatePlayingMediaDidChange(CanDelayNotification canDelayNotification)
 {
+    MediaProducerMediaStateFlags newState = internals().mainFrameMediaState;
+    protectedBrowsingContextGroup()->forEachRemotePage(*this, [&newState](auto& remotePageProxy) {
+        newState.add(remotePageProxy.mediaState());
```

Comment:
Does it make sense to logically OR a bunch of media state flags from disparate frames like this?

Presumably the answer is yes, since we were already doing the equivalent of this in WebContent for different documents:

```
void Page::updateIsPlayingMedia()
{
    MediaProducerMediaStateFlags state;
    forEachDocument([&](auto& document) {
        state.add(document.mediaState());
    });
```